signer/src/testing/requests.rs

```diff
@@ -0,0 +1,562 @@
+                    script_pubkey: sbtc::deposits::to_script_pubkey(
+                        deposit_script.clone(),
+                        reclaim_script.clone(),
+                    ),
+                },
+                TxOut {
+                    value: utxo.amount() - Amount::from_sat(amount_sats + fee),
+                    script_pubkey: self.depositor.address.script_pubkey(),
+                },
+            ],
+        };
+
+        regtest::p2tr_sign_transaction(&mut tx, 0, &[utxo], &self.depositor.keypair);
+
+        let create_req = CreateDepositRequest {
+            outpoint: OutPoint::new(tx.compute_txid(), 0),
+            deposit_script,
+            reclaim_script,
+        };
+
+        let info = create_req.validate_tx(&tx, false).unwrap();
+
+        let request = DepositRequest {
+            outpoint: info.outpoint,
+            max_fee: info.max_fee,
+            signer_bitmap: BitArray::ZERO,
+            amount: info.amount,
+            deposit_script: info.deposit_script.clone(),
+            reclaim_script: info.reclaim_script.clone(),
+            reclaim_script_hash: Some(TaprootScriptHash::from(&info.reclaim_script)),
+            signers_public_key: info.signers_public_key,
+        };
+
+        assert_eq!(tx.compute_txid(), request.outpoint.txid);
+
+        PreparedDeposit {
+            tx,
+            request,
+            info,
+            bitcoin: self.bitcoin,
+            emily: self.emily,
+            _bitcoin_txid: (),
+            _emily_deposit: (),
+        }
+    }
+}
+
+/// Provides specialized functionality for `DepositHelper` when:
+/// * `Bitcoin` is a [`Faucet`],
+/// * `Emily` is any type,
+/// * `Depositor` is any type that implements `Borrow<bitcoin::Address>`.
+impl<Emily, Depositor> DepositHelper<'_, Faucet, Emily, Depositor>
+where
+    Depositor: Borrow<bitcoin::Address>,
+{
+    /// Funds the inner `Depositor` from the faucet with the specified amount.
+    pub fn fund<Amt>(&self, amount: Amt) -> OutPoint
+    where
+        Amt: AsSatoshis,
+    {
+        self.bitcoin
+            .send_to(amount.as_satoshis(), self.depositor.borrow())
+    }
+}
+
+/// Provides specialized functionality for `DepositHelper` when:
+/// * `Bitcoin` implements `Borrow<bitcoincore_rpc::Client>`,
+/// * `Emily` is any type,
+/// * `Depositor` is a [`Recipient`].
+impl<'a, Bitcoin, Emily> DepositHelper<'a, Bitcoin, Emily, Recipient>
+where
+    Bitcoin: Borrow<bitcoincore_rpc::Client>,
+{
+    /// Returns a reference to the inner [`Recipient`] (depositor).
+    pub fn depositor(&self) -> &Recipient {
+        &self.depositor
+    }
+
+    /// Gets the balance of the inner [`Recipient`] (depositor) as a [`bitcoin::Amount`].
+    pub fn get_balance(&self) -> Amount {
+        self.depositor.get_balance(self.bitcoin.borrow())
+    }
+
+    /// Finds a suitable UTXO for the depositor that covers the specified
+    /// amount.
+    fn find_suitable_utxo<TxAmt, FeeAmt>(
```

Comment:
```suggestion
    /// amount.
    #[track_caller]
    fn find_suitable_utxo<TxAmt, FeeAmt>(
```
nit?

Reply:
This (private) fn returns a result instead of panicking so that sync vs async callers can handle according to their context.